torch/_dynamo/utils.py

```diff
@@ -774,9 +773,14 @@ def enum_repr(value):
     # Workaround repr(Enum) returning invalid global reference before python 3.11
     # https://peps.python.org/pep-0663/
     if sys.version_info < (3, 11):
-        return str(value)
+        enum_name = str(value)
     else:
-        return repr(value)
+        enum_name = repr(value)
+    # TODO(voz): Do I need to check upstream guard_source against GuardSource.LOCAL(_NN_MODULE)
+    # or pass L/G?
+    name, val = enum_name.split(".")
+    local_name = f'L["{name}"].{val}'
```

Comment:
This looks suspect

Reply:
The general problem is there is no reason to believe that the enum is actually in scope (either neither locals nor globals). In other words, we need to apply the same playbook as in https://github.com/pytorch/pytorch/pull/96940

Reply:
Hmm, I assumed this would be safe, but perhaps not. Alright, I can do something similar here.